### ISSUE TITLE
MacOS Javascript Error

### ISSUE THREAD

**Issue by purevertigo** [chrisknepper/android-messages-desktop#190]
I get the following error after opening the app.
![Screen Shot 2019-07-02 at 12 47 35](https://user-images.githubusercontent.com/6506461/60479279-9218c580-9cc8-11e9-812b-aef0caf4d339.jpg)

**Comment by timcappalli** [comment]
Also seeing this on 2.0.0 and 10.14.5

**Comment by greenbp** [comment]
I fixed this by recursively changing the permissions on /Applications/Android Messages.app to [my_user]:staff instead of root:wheel

```
MBP:~ bgreen$ sudo chown -R bgreen:staff /Applications/Android\ Messages.app/
```

**Comment by EXYZED** [comment]
A similar error on my end:

```
Uncaught Exception:
Error: EROFS: read-only file system, open '/private/var/folders/48/7zsx1c294zv9m42pc1lz8_m40000gn/T/AppTranslocation/E1322FF5-361E-4DC0-B60D-843D9D8FCB2F/d/Android Messages.app/Contents/Resources/resources/dictionaries/supported-languages.json'
    at Object.openSync (fs.js:436:3)
    at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:160:31)
    at Object.writeFileSync (fs.js:1168:35)
    at WriteStream.Array (/private/var/folders/48/7zsx1c294zv9m42pc1lz8_m40000gn/T/AppTranslocation/E1322FF5-361E-4DC0-B60D-843D9D8FCB2F/d/Android Messages.app/Contents/Resources/app.asar/app/webpack:/src/helpers/dictionary_manager.js:69:71)
    at WriteStream.emit (events.js:187:15)
    at lazyFs.open (internal/fs/streams.js:273:12)
    at FSReqWrap.oncomplete (fs.js:141:20)
```

![Screenshot 2019-07-16 at 12 36 02](https://user-images.githubusercontent.com/18582414/61291406-5ac90f00-a7c6-11e9-9f49-a64ef11161ad.png)

**Comment by purevertigo** [comment]
> I fixed this by recursively changing the permissions on /Applications/Android Messages.app to [my_user]:staff instead of root:wheel
> 
> ```
> MBP:~ bgreen$ sudo chown -R bgreen:staff /Applications/Android\ Messages.app/
> ```

That fixed it!

Hopefully, this fix can be incorporated into the next update.

**Comment by chrisknepper** [comment]
@purevertigo @greenbp @EXYZED @timcappalli 

Hey all, this should now be fixed in v3.0.0 which I just released. This bug was a side effect of me not storing certain files in the right directory.

For some background, when the app opens, it asks your OS what language it should use and then downloads the spelling dictionaries for that language (if available) as well as a JSON file which lists the languages available to download. It is done this way because if I were to include these files in the app, it would about double the size of the output files (.exe/.dmg/.deb etc).

The problem was I had programmed the downloads to go into the Electron `resources` directory (where things like the icons are), but Electron intends this directory to be read-only. The correct solution is to use the `userData` directory provided by Electron which is safe to write to.

Please download v3.0.0 and file a new issue if you encounter something wrong. Apologies that this took so long and thanks for using the app!
